openml-h2o/src/main/java/com/feedzai/openml/h2o/H2OModelCreator.java

```diff
@@ -133,13 +144,44 @@ public AbstractClassificationH2OModel loadModel(final Path modelPath,
             );
         }
 
+        validateSchema(genModel, schema);
         final AbstractClassificationH2OModel resultingModel = createModel(modelPath, schema, genModel, closeable);
         ClassificationValidationUtils.validateClassificationModel(schema, resultingModel);
 
         logger.info("Model loaded successfully.");
         return resultingModel;
     }
 
+    /**
+     * Performs a validation from the H2O model against the schema provided.
+     *
+     * <p>
+     *     Note that the field names stored in the {@link IGeneratedModel provided model} might not necessarily include the target variable
+     *     (i.e. response field/column).
+     *     As such, this validation step is adaptation in that sense (i.e. it allows the expected schema from the provided model to either have or
```

Comment:
I know that this approach is a bit too relaxed for a normal validation step, but it's H2O I guess 🤷‍♂️